protobuf/pg_query.proto

```diff
@@ -4108,3 +4108,44 @@ enum Token {
   MODE_PLPGSQL_ASSIGN3 = 777;
   UMINUS = 778;
 }
+
+
+message SummaryResult {
+  enum Context {
+    None = 0;
+    Select = 1;
+    DML = 2;
+    DDL = 3;
+    Call = 4;
+  }
+
+  message Table {
+    string table = 1;
+    Context context = 2;
+  }
+  // tables: Vec<(String, Context)>
+  repeated Table tables = 1;
+
+  // aliases: HashMap<String, String>
+  map<string, string> aliases = 2;
+
+  // cte_names: Vec<String>
+  repeated string cte_names = 3;
+
+  // functions: Vec<(String, Context)>
+  message Function {
+    string name = 1;
```

Comment:
As discussed, would be good to also add `function_name` and `schema_name` here, just to make it easier for the caller to work with the result (and avoid an unnecessary split on the qualified name).

To get those values:
- `function_name` is the last entry in the `funcname` list
- `schema_name` is the second to last entry in the `funcname` list (and may not be present)

(there could also be a third entry, which is the catalog or database name, but we can ignore that)